### ISSUE TITLE
python 2.7 exceptions are not printed.

### ISSUE THREAD

**Issue by digi604** [rethinkdb/rethinkdb#4114]
The actual exception message is not printed in python 2.7.

**Comment by digi604** [comment]
There also seams to be a problem with celerys LoggingProxy:

File "/Users/patricklauber/Documents/workspace/pitcher-tasks/env/lib/python2.7/site-packages/rethinkdb/errors.py", line 18, in convertForPrint
    return inputString.encode(sys.stdout.encoding or 'utf-8',
AttributeError: 'LoggingProxy' object has no attribute 'encoding'

I would say the current Exceptions are completly broken on python 2.7

**Comment by coffeemug** [comment]
@digi604 -- could you give a little more detail? The following would help:

- What RethinkDB query are you running?
- How are your running it? Through a Python REPL, some other way, etc.?
- What output are you getting?

**Comment by danielmewes** [comment]
@digi604 Does this example work for you or do you get the same error?

```
$ python2.7
Python 2.7 (r27:82500, Jul 22 2010, 19:16:27) 
[GCC 4.4.1] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import rethinkdb as r;
>>> r.connect(host='localhost').repl()
<rethinkdb.net.Connection object at 0x7f7860b3ea90>
>>> r.expr(r.literal()).run()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/rethinkdb/ast.py", line 108, in run
    return c._start(self, **global_opt_args)
  File "/usr/local/lib/python2.7/site-packages/rethinkdb/net.py", line 159, in _start
    return self._send_query(query, term, global_opt_args)
  File "/usr/local/lib/python2.7/site-packages/rethinkdb/net.py", line 229, in _send_query
    raise RqlRuntimeError(message, term, frames)
rethinkdb.errors.RqlRuntimeError: Stray literal keyword found: literal is only legal inside of the object passed to merge or update and cannot nest inside other literals. in:
r.literal()
^^^^^^^^^^^
>>> 
```

I wonder where that `LoggingProxy` object is coming from. Are you using the logging-proxy package? We don't seem to be using it in our driver (at least not directly).

@larkost @Tryneus any ideas on this?

**Comment by Tryneus** [comment]
It looks like `sys.stdout` has been replaced by a `LoggingProxy` object, which doesn't have the same semantics as a `file` object.

**Comment by danielmewes** [comment]
Oh I see. I think we should handle the case where `sys.stdout.encoding` doesn't exist.

I found this thread which is about a related issue where LoggingProxy doesn't provide a proper `fileno()` on `stdout` https://github.com/celery/celery/issues/928 .

The conclusion seems to be that Python doesn't require anything else but a `write()` method on `sys.stdout`, though the documentation is a bit ambiguous (https://docs.python.org/2/library/sys.html#sys.stdin)

**Comment by larkost** [comment]
This has been fixed in #4764.
